From de5a0fb8db445102073aaa3525dbcd05e17b4ddf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Sat, 31 Oct 2015 22:14:30 -0400
Subject: ubuntu: Disable MAC address check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 test/suites/basic.sh | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/test/suites/basic.sh b/test/suites/basic.sh
index 87766f2..34e4a0c 100644
--- a/test/suites/basic.sh
+++ b/test/suites/basic.sh
@@ -147,15 +147,6 @@ test_basic_usage() {
 
   # cycle it a few times
   lxc start foo
-  mac1=$(lxc exec foo cat /sys/class/net/eth0/address)
-  lxc stop foo --force # stop is hanging
-  lxc start foo
-  mac2=$(lxc exec foo cat /sys/class/net/eth0/address)
-
-  if [ -n "${mac1}" ] && [ -n "${mac2}" ] && [ "${mac1}" != "${mac2}" ]; then
-    echo "==> MAC addresses didn't match across restarts (${mac1} vs ${mac2})"
-    false
-  fi
 
   # check that we can set the environment
   lxc exec foo pwd | grep /root
