From 5b7ef7f863383cdf217a53b29c069a102f5e2ea4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Tue, 29 Sep 2015 23:18:32 -0400
Subject: tests: Don't fail when GOPATH isn't set
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 test/main.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/main.sh b/test/main.sh
index cca98f7..6261e7a 100755
--- a/test/main.sh
+++ b/test/main.sh
@@ -1,5 +1,5 @@
 #!/bin/sh -eu
-export PATH=${GOPATH}/bin:${PATH}
+[ -n "${GOPATH:-}" ] && export PATH=${GOPATH}/bin:${PATH}
 
 if [ -n "${LXD_DEBUG:-}" ]; then
   set -x
@@ -169,7 +169,7 @@ cleanup() {
       echo "failed test: ${TEST_CURRENT}"
     fi
 
-    echo "To poke around, use:\n LXD_DIR=${LXD_DIR} sudo -E ${GOPATH}/bin/lxc COMMAND --config ${LXD_CONF}"
+    echo "To poke around, use:\n LXD_DIR=${LXD_DIR} sudo -E ${GOPATH:-}/bin/lxc COMMAND --config ${LXD_CONF}"
     read -p "Tests Completed (${TEST_RESULT}): hit enter to continue" x
   fi
 
