From 6a0571ad34c329cdf0d24e83341d91cd3dba032c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Sat, 25 Jul 2015 20:17:40 +0200
Subject: Fix container rootfs mode in split images
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Closes #895

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/images.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxd/images.go b/lxd/images.go
index 8e5edae..d1941ae 100644
--- a/lxd/images.go
+++ b/lxd/images.go
@@ -95,7 +95,7 @@ func untarImage(imagefname string, destpath string) error {
 
 	if shared.PathExists(imagefname + ".rootfs") {
 		rootfsPath := fmt.Sprintf("%s/rootfs", destpath)
-		err = os.MkdirAll(rootfsPath, 0700)
+		err = os.MkdirAll(rootfsPath, 0755)
 		if err != nil {
 			return fmt.Errorf("Error creating rootfs directory")
 		}
