From 441b170299158ffc7a76dc8fec6e80e32e956329 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Thu, 15 Oct 2015 09:54:23 -0700
Subject: Fix non-interactive mode detection
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxc/exec.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxc/exec.go b/lxc/exec.go
index 9e6da22..0b14613 100644
--- a/lxc/exec.go
+++ b/lxc/exec.go
@@ -125,7 +125,7 @@ func (c *execCmd) run(config *lxd.Config, args []string) error {
 		env[pieces[0]] = value
 	}
 
-	cfd := syscall.Stdout
+	cfd := syscall.Stdin
 	var oldttystate *terminal.State
 	interactive := terminal.IsTerminal(cfd)
 	if interactive {
