From a80bc6c2c2652a8a1db6714e82c41a92dbcd5065 Mon Sep 17 00:00:00 2001
From: Tycho Andersen <tycho.andersen@canonical.com>
Date: Wed, 9 Dec 2015 08:40:52 -0700
Subject: remove extra debugging statement

Signed-off-by: Tycho Andersen <tycho.andersen@canonical.com>
---
 scripts/lxd-images | 1 -
 1 file changed, 1 deletion(-)

diff --git a/scripts/lxd-images b/scripts/lxd-images
index d8f96c1..0f01d82 100755
--- a/scripts/lxd-images
+++ b/scripts/lxd-images
@@ -99,7 +99,6 @@ class LXD(object):
 
         # Create our workdir
         self.workdir = tempfile.mkdtemp()
-        print("workdir is " + self.workdir)
 
     def cleanup(self):
         if self.workdir:
