From 502b9d81b79954089d2ce24a63dd406f2c9dff98 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Wed, 9 Dec 2015 15:47:20 -0500
Subject: Apparently IsMountPoint is never very reliable with bind-mounts...
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxd/container_lxc.go | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/lxd/container_lxc.go b/lxd/container_lxc.go
index c09da7e..518360e 100644
--- a/lxd/container_lxc.go
+++ b/lxd/container_lxc.go
@@ -2624,13 +2624,8 @@ func (c *containerLXC) removeDiskDevices() error {
 			continue
 		}
 
-		// Unmount the host side
-		if shared.IsMountPoint(filepath.Join(c.DevicesPath(), f.Name())) {
-			err = syscall.Unmount(filepath.Join(c.DevicesPath(), f.Name()), syscall.MNT_DETACH)
-			if err != nil {
-				return err
-			}
-		}
+		// Always try to unmount the host side
+		_ = syscall.Unmount(filepath.Join(c.DevicesPath(), f.Name()), syscall.MNT_DETACH)
 
 		// Remove the entry
 		err := os.Remove(filepath.Join(c.DevicesPath(), f.Name()))
